rmi/servercode/TransImpl/HashVault.java

```diff
@@ -60,6 +61,48 @@ public void serialize_in() {
     try {
       // Initialize File/Serailization Streams
       FileInputStream file_pipe = new FileInputStream("shadowVault.ser");
+      ObjectInputStream object_pipe = new ObjectInputStream(file_pipe);
+
+      // Serialize stable Vault into runtime Vault
+      this.vault = (HashMap<Integer, RMHashtable>) object_pipe.readObject();
+
+      // Closes Streams
+      file_pipe.close();
+      object_pipe.close();
+    }
+    catch (IOException e) {
+      e.printStackTrace();
+    }
+    catch (ClassNotFoundException e) {
+      e.printStackTrace();
+    }
+  }
+
+  // Want to store the vault to its stable file? Call this function
+  public void serialize_out_temp(int transactionId) {
```

Comment:
serialize_out() and serialize_in() will just write or read from the master record.

serialize_out_temp(int) and serialize_in_temp(int) will write and read from a temporary transaction shadow